colexecdisk: flake in TestExternalSortMemoryAccounting

Found here:

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/7502751?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
Failed
=== RUN   TestExternalSortMemoryAccounting
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/53d72e801f20c60e0a7b0e08d0b2dc41/logTestExternalSortMemoryAccounting2342852196
    test_log_scope.go:79: use -show-logs to present logs inline
    external_sort_test.go:140:
          Error Trace:  /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10211/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/colexec/colexecdisk/colexecdisk_test_/colexecdisk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/colexec/colexecdisk/external_sort_test.go:140
          Error:        "3" is not greater than or equal to "4"
          Test:         TestExternalSortMemoryAccounting
          Messages:     didn't create enough partitions: actual 3, min expected 4
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/53d72e801f20c60e0a7b0e08d0b2dc41/logTestExternalSortMemoryAccounting2342852196
--- FAIL: TestExternalSortMemoryAccounting (1.27s)
```

cc @yuzefovich @michae2 for triage.

Jira issue: CRDB-21457

--- a/pkg/sql/colexec/colexecdisk/external_sort_test.go
+++ b/pkg/sql/colexec/colexecdisk/external_sort_test.go
@@ -50,7 +50,6 @@ func TestExternalSortMemoryAccounting(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 91850) // flaky test
 	skip.UnderStress(t, "the test is very memory-intensive and is likely to OOM under stress")
 
 	ctx := context.Background()
@@ -134,12 +133,14 @@ func TestExternalSortMemoryAccounting(t *testing.T) {
 	// numInMemoryBufferedBatches (first merge = 2x, second merge = 3x, third
 	// merge 4x, etc, so we expect 2*numNewPartitions-1 partitions).
 	expMaxTotalPartitionsCreated := 2*numNewPartitions - 1
-	// Because of the fact that we are creating partitions slightly larger than
-	// memoryLimit in size and because of our "after the fact" memory
-	// accounting, we might create less partitions than maximum defined above
-	// (e.g., if numNewPartitions is 4, then we will create 3 partitions when
-	// batch size is 3).
-	expMinTotalPartitionsCreated := numNewPartitions - 1
+	// Since we are creating partitions slightly larger than memoryLimit in size
+	// and because of our "after the fact" memory accounting, we might create
+	// fewer partitions than the target (e.g., if numNewPartitions is 4, then we
+	// will create 3 partitions when batch size is 3). However, we might not
+	// even create numNewPartitions-1 in edge cases (due to how we grow
+	// coldata.Bytes.buffer when setting values), so we opt for a sanity check
+	// that at least two partitions were created that must always be true.
+	expMinTotalPartitionsCreated := 2
 	require.GreaterOrEqualf(t, numPartitionsCreated, expMinTotalPartitionsCreated,
 		"didn't create enough partitions: actual %d, min expected %d",
 		numPartitionsCreated, expMinTotalPartitionsCreated,

